libsolidity/ast/ASTJsonConverter.cpp

```diff
@@ -108,21 +103,56 @@ Json::Value const& ASTJsonConverter::json()
 	return m_astJson;
 }
 
+bool ASTJsonConverter::visit(SourceUnit const&)
+{
+	Json::Value children(Json::arrayValue);
+
+	m_astJson["name"] = "root";
```

Comment:
I though the first element should be `root`. If not, I'll rename it.

`addJsonNode()` appends to the top node `m_jsonNodePtrs.top()->append(node);`. There's no nodes in the stack before the root node.